### DISCUSSION TITLE
ACME version ... Windows only?

### DISCUSSION THREAD

**Discussion by cminter** [jkotlinski/durexforth#333] [Q&A] [answered]
What ACME version is being used?  Are you building on Windows only?

I was able to build **on Linux** a couple years ago (and had compiled 'acme' from acme091src.tar.bz2).  But trying to build today (eeca09a9911dfc5e32de89dde060036166d44200) gives errors...

    Error - File durexforth.asm, line 126 (Zone <untitled>): There's more than one character.
    Error - File durexforth.asm, line 126 (Zone <untitled>): Macro not defined (or wrong signature).
    Error - File durexforth.asm, line 109 (Macro BACKLINK): Garbage data at end of statement.
    Serious error - File durexforth.asm, line 111 (Macro BACKLINK): Value not yet defined.
    make: *** [Makefile:35: durexforth.prg] Error 1

Thanks, Corey

**Comment by Whammo** [comment]
Did you have any luck?

**Reply by cminter** [reply]
I was really first just trying to hear from anyone regarding what version is expected to work before I did anything.

I see:
+ An [SVN repo on Sourceforge](https://sourceforge.net/projects/acme-crossass/) which released 0.97 in 2020-07 but only shows Windows and Mac releases.
+ A [GitHub repo](https://github.com/meonwax/acme) that shows 0.96.4 but has only updates two years ago.
+ An [Ubuntu package](https://launchpad.net/ubuntu/+source/acme/1:0.96.4-5) also 0.96.4 but I don't use Ubuntu.

Of course I can just try to build from sources on either Sourceforge source or Github.  But I'd hope to focus on the preferred active repo.

**Comment by Whammo** [comment, marked as answer]
These videos have a description of the process of compilation.

[https://www.youtube.com/watch?v=TXIDqptXmiM](url)
[https://www.youtube.com/watch?v=1oZztCmC8kc](url)

**Reply by cminter** [reply]
Oh I watched those videos before.  Good stuff.  At the time, I may have skipped over the ACME part.

I liked the random number generator details.

**Reply by Whammo** [reply]
I haven't compiled it, myself. If you get around to it, attach the d64 here, if you please.

**Reply by cminter** [reply]
OK yea let me revisit what I had from following along with his demo.

FYI:  I really wanted to try out the RS232 stuff you and others were working on in [another thread](https://github.com/jkotlinski/durexforth/discussions/330).

**Reply by Whammo** [reply]
Then I suggest you compile my fork. I'd like to see people getting use of it.

**Reply by cminter** [reply]
hey sorry I keep editing my post and edited a lot of what I wrote in the last one.

**Reply by cminter** [reply]
> I haven't compiled it, myself. If you get around to it, attach the d64 here, if you please.

Here is a [d64 of source based on his guessnum demo](https://drive.google.com/file/d/140Va62VMcwqD4HTLOJ5LtqrvLrNKJP46/view?usp=sharing).  It was from a couple years ago.  I might have made some slight changes.

**Reply by Whammo** [reply]
I was looking for a compilation of the current state. I compiled it myself before, but the toolchain is pretty complex. I think I just compiled it straight from assembly with acme then used DirMaster or something to put together a d64.

**Reply by cminter** [reply]
Sure.  I compiled ACME 0.97 (from Sourceforge src).  And then I was able to build to a **d64** (from eeca09a9911dfc5e32de89dde060036166d44200).
[durexforth-geeca09a-20210423.d64](https://drive.google.com/file/d/13w0FCAhqpwDu7UunWvXJLl3CRybI0N8I/view?usp=sharing)
THIS IS NOT REALLY VALID/COMPLETE - I hadn't run the deploy step, I'll followup later

**Reply by cminter** [reply]
And here is your Whammo repo (from c08270e6927b5346f2913a743aa2e56b4196e2d5)
[durexforth-gc08270e-20210423.d64](https://drive.google.com/file/d/1EriN51m7URNKkGd5pHpO56SQFucY1ELL/view?usp=sharing)
THIS IS NOT REALLY VALID/COMPLETE - I hadn't run the deploy step, I'll followup later

**Reply by Whammo** [reply]
Thank you!

**Reply by cminter** [reply]
Results from `make deploy`.  This is the deploy directory, so the naming shows the last tag (not number commits beyond)

from jkotlinski eeca09a9911dfc5e32de89dde060036166d44200:  [durexforth-geeca09a-20210423.zip](https://drive.google.com/file/d/1I_xS3Gx8K7iX5iBDqHff0McZt5ASWWEW/view?usp=sharing)
from Whammo c08270e6927b5346f2913a743aa2e56b4196e2d5:  [durexforth-Whammo-gc08270e-20210423.zip](https://drive.google.com/file/d/1zNI6nO9Q88APE6LTFOUk2yAvVJsTcfsZ/view?usp=sharing)

**Reply by Whammo** [reply]
Smitty- if you're paying attention- grab this and see what you think.

**Reply by cminter** [reply]
@Whammo , I see you had updates...
from Whammo 679fd4e4479fe498ba71aab7eca12eaf450a9b47:  [durexforth-Whammo-g679fd4e-20210423.zip](https://drive.google.com/file/d/1UjeVe9z2VyLfWvZPQ1Xp1ErAzmkVny1B/view?usp=sharing)

**Reply by Whammo** [reply]
Thanks, let me check this iteration.

**Comment by jkotlinski** [comment]
Hi Corey, sorry for late reply. I use ACME 0.97. If you use Linux, it should be easy to build ACME yourself from source.

**Reply by cminter** [reply]
No problem.  I went ahead and tried it about 15 minutes ago.  It works fine as you just wrote.
